src/di.rs

```diff
@@ -172,27 +146,49 @@ where
             .downcast::<V>()
             .expect("Checked by .unwrap_or_else()")
     }
+
+    fn available_types(&self) -> String {
+        let mut list = String::new();
+
+        for dep in self.map.values() {
+            writeln!(list, "    {}", dep.type_name).unwrap();
+        }
+
+        list
+    }
 }
 
-impl<V, S> DependencySupplier<V> for Arc<S>
-where
-    S: DependencySupplier<V>,
-{
-    fn get(&self) -> Arc<V> {
-        self.deref().get()
+impl Debug for DependencyMap {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        f.debug_struct("DependencyMap").finish()
```

Comment:
I don't remember why a manual implementation of `Debug` was chosen. Maybe just `#[derive(Debug)]`?

Reply:
You can't just `derive(Debug)`, because `dyn Any + Send + Sync` does not implement `Debug`.

Although mh. You should probably derive it on `DependencyMap` and then do a custom impl for `Dependency`.

Reply:
Done.